cmd/support/main.go

```diff
@@ -381,3 +443,58 @@ func getServiceLog(service string, dest string) {
 	}
 
 }
+
+// redactPasswords removes any occurrences of a passwd or password set in yaml/cloud-config files usually
+func redactPasswords(input []byte) []byte {
+	passwd := regexp.MustCompile(`(passwd)([\s=\t\:]{1,3})(.*)`)
+	password := regexp.MustCompile(`(password)([\s=\t\:]{1,3})(.*)`)
+	redacted := passwd.ReplaceAll(input, []byte("$1$2*****"))
+	redacted = password.ReplaceAll(redacted, []byte("$1$2*****"))
+	return redacted
+}
+
+// copyFilesInDir copies all files in the source dir to destination path, keeping the name. Follows dirs inside dirs.
+func copyFilesInDir(src, dest string) {
+	info, err := os.Lstat(src)
+	if err != nil {
+		logrus.Warnf("Error opening %s", src)
+		return
+	}
+	switch {
+	case info.IsDir():
+		copyDir(src, dest)
+	default:
+		copySingleFile(src, dest)
+	}
+}
+
+func copySingleFile(src string, dest string) {
+	logrus.Debugf("Copying %s into %s", src, dest)
+	original, err := os.ReadFile(src)
+	if err != nil {
+		logrus.Warnf("Failed to read: %s", err)
+		return
+	}
+	redacted := redactPasswords(original)
+	err = os.MkdirAll(filepath.Dir(dest), os.ModeDir|os.ModePerm)
+	if err != nil {
+		logrus.Warnf("Failed to create dir: %s", err)
+		return
+	}
+	err = os.WriteFile(dest, redacted, os.ModePerm)
+	if err != nil {
+		logrus.Warnf("Failed to write: %s", err)
+		return
+	}
+}
+
+func copyDir(srcdir string, destdir string) {
+	contents, err := os.ReadDir(srcdir)
+	if err != nil {
+		return
```

Comment:
can add a log entry in case it fails?

Reply:
same as above, dont care if it fails, its not supposed to give errors to the user, just tryu to get as much info as possible otherwise you get an error spouting util on your already broken system. Bad vibes :)